src/extension/geofence-server/src/test/java/org/geoserver/geofence/integration/GeofenceAccessManagerIntegrationTest.java

```diff
@@ -708,6 +729,313 @@ public void testAdminRule_WorkspaceAccessLimits_Username_based_rule() {
         assertAdminAccess(sfAdmin, sf, true);
     }
 
+    @Test
+    public void testAllowedAreaLayerInTwoGroupsEnlargementWithSingle() throws Exception {
+        // tests that when a Layer is directly accessed for WMS request
+        // if it is belonging to more then one LayerGroup, if one of the container
+        // LayerGroup is SINGLE, there will be no allowedArea in the final
+        // filter.
+        Long idRule = null;
+        LayerGroupInfo group1 = null;
+        LayerGroupInfo group2 = null;
+        try {
+            Authentication user = getUser("anonymousUser", "", "ROLE_ANONYMOUS");
+            login("admin", "geoserver", new String[] {"ROLE_ADMINISTRATOR"});
+            Catalog catalog = getCatalog();
+
+            LayerInfo places = catalog.getLayerByName(getLayerId(MockData.NAMED_PLACES));
+            LayerInfo forests = catalog.getLayerByName(getLayerId(MockData.FORESTS));
+
+            group1 =
+                    createsLayerGroup(
+                            catalog,
+                            "group21",
+                            LayerGroupInfo.Mode.SINGLE,
```

Comment:
You can probably use some static imports to reduce the line length, and the catalog can be grabbed inside the method (so remove a parameter) to avoid having all this wrapping, and keep the method call in a single line. When methods calls wrap lke this the code is either too nested, or the method has too many parameters.